src/main/java/org/broadinstitute/hellbender/utils/runtime/ProcessControllerBase.java

```
@@ -0,0 +1,157 @@
+    protected boolean destroyed = false;
+
+    // Useful for debugging if background threads have shut down correctly
+    private static int nextControllerId = 0;
+    protected final int controllerId;
+
+    public ProcessControllerBase() {
+        synchronized (running) {
+            controllerId = nextControllerId++;
+        }
+    }
+
+    /**
+     * @return The set of still running processes.
+     */
+    public static Set<ProcessControllerBase<?>> getRunning() {
+        synchronized (running) {
+            return new LinkedHashSet<>(running);
+        }
+    }
+
+    /**
+     * Executes a command line program with the settings and waits for it to return,
+     * processing the output on a background thread.
+     *
+     * @param settings Settings to be run.
+     * @return The output of the command.
+     */
+    protected Process launchProcess(ProcessSettings settings) {
+        Utils.validate(!destroyed, "This controller was destroyed");
+
+        final ProcessBuilder builder = new ProcessBuilder(settings.getCommand());
+        builder.directory(settings.getDirectory());
+
+        final Map<String, String> settingsEnvironment = settings.getEnvironment();
+        if (settingsEnvironment != null) {
+            final Map<String, String> builderEnvironment = builder.environment();
+            builderEnvironment.clear();
+            builderEnvironment.putAll(settingsEnvironment);
+        }
+
+        builder.redirectErrorStream(settings.isRedirectErrorStream());
+
+        // Start the process running.
+        try {
+            process = builder.start();
+            running.add(this);
+        } catch (IOException e) {
+            final String message = String.format("Unable to start command: %s\nReason: %s",
+                    StringUtils.join(builder.command(), " "),
+                    e.getMessage());
+            throw new GATKException(message);
```

you should propagate the underlying exception here

---

Yup. Done.